library/alloc/src/vec.rs

```diff
@@ -2797,6 +2797,13 @@ where
 ///
 /// This `struct` is created by the `into_iter` method on [`Vec`] (provided
 /// by the [`IntoIterator`] trait).
+///
+/// # Example
+///
+/// ```
+/// let v = vec![0, 1, 2];
+/// let iter = v.into_iter();
```

Comment:
Maybe say explicitly these are the same type?
```suggestion
/// let iter: std::vec::IntoIter<_> = v.into_iter();
```

Reply:
Not sure what value it adds to the example, Given it's a vector it's assumed to be items of the same type or I missed something?

Reply:
This is documentation on `IntoIter`, so this tests that `.into_iter()` does in fact return `IntoIter`.